File: backend/Entities/Expense.cs
using System;

namespace backend.Entities;

public enum ExpenseOption
{
    HPD_PRSTEN,
    OTHER
}

public class Expense
{
    public int Id { get; set; }
    public ExpenseOption Option { get; set; }
    public decimal Amount { get; set; }
    public DateOnly Date { get; set; }
    public string? Description { get; set; }
}

File: frontend/src/types/expense.ts
import { Dayjs } from "dayjs";

export enum ExpenseOption {
	HPD_PRSTEN = 0,
	OSTALO = 1,
}

export type ExpenseOptionValue = ExpenseOption.HPD_PRSTEN | ExpenseOption.OSTALO;

export interface Expense {
	option: number;
	amount: number | null;
	date: Dayjs | null;
	description: string;
}

export interface GridExpense extends Omit<Expense, "option" | "amount" | "date"> {
	option: string;
	amount: string;
	date: string;
}
